src/Filesystem/FilesystemInterface.php

```diff
@@ -120,7 +120,7 @@ public function symlink(string $originDir, string $targetDir, bool $copyOnWindow
      *
      * @throws IOException When file type is unknown
      */
-    public function mirror(string $originDir, string $targetDir, \Traversable $iterator = null, array $options = []);
+    public function mirror(string $originDir, string $targetDir, ?\Traversable $iterator = null, array $options = []);
```

Comment:
PHP 8.4 deprecates implicitly nullable types.
It is also worth adding this change to the implementing class.